### DISCUSSION TITLE
"Device not migrated" even though String Descriptors are sent ok

### DISCUSSION THREAD

**Discussion by deeebug** [hathach/tinyusb#1775] [Q&A]
Hello,

I'm attempting to make a gamepad that works properly with the PS3, and as a PS controller on Windows.
For this, I'm looking at the UnoJoy project. I've managed to copy the descriptors and whatnot which got it working on the Playstation. It does fine on Linux too. The problem exists only in Windows. And it seems to occur at the String Descriptor exchange. It results in problem being displayed in the device property events as "Device not migrated".

Here's the image of the Wireshark captures side by side.
Left: Works, original UnoJoy
Right: "Device not migrated", devices works but PS button doesn't on Windows, UnoJoy copy
![image](https://user-images.githubusercontent.com/77402236/205497029-35612977-ffba-4c12-b9ea-cef41f0566fa.png)

In the image, the left one is the UnoJoy controller that works nicely which is the reference. Right one is what I've manged to get going. It all works but it seems to fail during the String Descriptor part. From the get go, in the first String Descriptor request, it asks for the index `0x02` also with a strange `wLength` of `2050`. The device does respond to it along with the others but Windows keeps asking for it repeatedly until it doesn't.
There's also an extra interrupt being sent in the middle while the UnoJoy (left) one waits until the String Descriptor part completes, but I'm not sure if that's causing the problem. If it is, how to synchronize that?

Completely out of ideas on what to do next. Please suggest what needs to be looked at.
I've also attached the PCAP files.
[PCAPs.zip](https://github.com/hathach/tinyusb/files/10148787/PCAPs.zip)

Board: RP2040 Raspberry Pi 2040

**Comment by hathach** [comment]
enable tinyusb log=2, it could give some hint
